tests/views/test_survey_response_view.py

```diff
@@ -0,0 +1,36 @@
+from application.controllers.survey_response import UPLOAD_SUCCESSFUL
+from requests.models import Response
+from six import BytesIO
+from tests.test_client import TestClient
+from unittest.mock import patch
+
+
+class TestSurveyResponseView(TestClient):
+    """ Survey response unit tests"""
+
+    def test_add_survey_response_success(self):
+
+        # Given a file with mocked micro service calls to case, collection and case
+        data = dict(file=(BytesIO(b'upload_test'), 'upload_test.xls'))
+
+        mock_response = Response()
+        mock_response.status_code = 200
+        mock_case = {'caseGroup': {'sampleUnitRef': 'sampleUnitRef', 'collectionExerciseId': 'collectionExerciseId'}}
+        mock_collection = {'exerciseRef': 'test', 'surveyId': 'test'}
+
+        with patch('application.controllers.survey_response.SurveyResponse._get_case', return_value=mock_case), \
```

Comment:
Would it have been better to mock the service call instead of the _private_ method?

Reply:
Can I mock variables in the patch, as the service call is used both for case and collection?

Reply:
You can do something like https://stackoverflow.com/a/24897297/4695665 where the first time it's called it will return the case, second the collection